src/software/ai/navigator/path_planner/hrvo/brute_force_nearest_neighbor_search.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <vector>
+
+/**
+ * Finds the neighbors of a candidate based on a comparator and radius
+ *
+ * @tparam T generic type for the candidate and its neighbors
+ * @tparam Func type for the comparator
+ * @param candidate cannot be null
+ * @param neighbors a list of T objects neighbors, which may contain the candidate object
+ * itself
+ * @param radius the maximum distance from a candidate object
+ * @param comparator function that measures the square distance between a candidate and a
+ * neighbor
+ * @return a list representing a subset of neighbors
+ */
+template <class T, typename Func>
+std::vector<T> nearestNeighbours(const T& candidate, const std::vector<T>& neighbors,
+                                 double radius, Func comparator)
+{
+    std::vector<T> subset;
+    for (const T& neighbor : neighbors)
+    {
+        if (comparator(candidate, neighbor) < radius * radius && candidate != neighbor)
```

Comment:
Actually I thought about it a bit more and to make this function a bit more generalized, we should just use `radius`, instead of squared radius in this implementation. It will be then up to the caller to pass the squared radius and calculate squared radius of two `T`s inside `comparator`.

Reply:
Also, since we use the name `neighbours` for the array (implying that its neighbours and doesn't include _candidate_) it's probably better if we don't include `&& candidate != neighbor`. That would also be up to the caller to afterwards delete the `candidate` from the returned `subset`, or leave it. In the case that the `neighbours` list _doesn't_ have `candidate` in it, there should also be a slight performance boost.

Reply:
> Actually I thought about it a bit more and to make this function a bit more generalized, we should just use `radius`, instead of squared radius in this implementation. It will be then up to the caller to pass the squared radius and calculate squared radius of two `T`s inside `comparator`.

Sure this sounds good, we will leave it up to the caller if they want to compare square radius or non-square radius. 

> Also, since we use the name neighbours for the array (implying that its neighbours and doesn't include candidate) it's probably better if we don't include && candidate != neighbor. That would also be up to the caller to afterwards delete the candidate from the returned subset, or leave it. In the case that the neighbours list doesn't have candidate in it, there should also be a slight performance boost.

Done. Similar idea to above, will leave it up to the caller whether or not they want too include/remove the candidate in the subset, if `neighbours` list have `candidate` in it.